Sources/Danger/GitDiff.swift

```diff
@@ -0,0 +1,169 @@
+            return .created(addedLines: hunks.flatMap { $0.lines.map(\.text) })
+        case .deleted:
+            return .deleted(deletedLines: hunks.flatMap { $0.lines.map(\.text) })
+        case .modified:
+            return .modified(hunks: hunks)
+        case let .renamed(oldPath: oldPath):
+            return .renamed(oldPath: oldPath, hunks: hunks)
+        }
+    }
+
+    var description: String {
+        hunks.map(\.description).joined(separator: "\n")
+    }
+}
+
+extension FileDiff {
+    struct ParsedHeader: Equatable {
+        let filePath: String
+        let change: ChangeType
+    }
+}
+
+extension FileDiff.ParsedHeader {
+    enum ChangeType: Equatable {
+        case created
+        case deleted
+        case renamed(oldPath: String)
+        case modified
+    }
+}
+
+extension FileDiff {
+    enum Changes: Equatable {
+        case created(addedLines: [String])
+        case deleted(deletedLines: [String])
+        case modified(hunks: [Hunk])
+        case renamed(oldPath: String, hunks: [Hunk])
+    }
+
+    struct Hunk: Equatable, CustomStringConvertible {
+        public let oldLineStart: Int
+        public let oldLineSpan: Int
+        public let newLineStart: Int
+        public let newLineSpan: Int
+        public let lines: [Line]
+        var description: String {
+            "@@ -\(oldLineStart),\(oldLineSpan) +\(newLineStart),\(newLineSpan) @@" + lines.map(\.description).joined(separator: "\n")
+        }
+    }
+
+    struct Line: Equatable, CustomStringConvertible {
+        let text: String
+        let changeType: ChangeType
+        var description: String {
+            switch changeType {
+            case .added:
+                return "+" + text
+            case .removed:
+                return "-" + text
+            case .unchanged:
+                return " " + text
+            }
+        }
+    }
+}
+
+extension FileDiff.Line {
+    enum ChangeType: Equatable {
+        case added
+        case removed
+        case unchanged
+    }
+}
+
+extension StringProtocol {
+    func deletingPrefix(_ prefix: String) -> String {
+        guard hasPrefix(prefix) else { return String(self) }
+        return String(dropFirst(prefix.count))
+    }
+}
+
+struct DiffParser {
+    func parse(_ diff: String) -> [FileDiff] {
+        diff.components(separatedBy: "diff --git ").compactMap { fileDiff in
+            let headerAndHunks = fileDiff.components(separatedBy: "@@")
+            guard let header = headerAndHunks.first,
+                !header.isEmpty else {
+                return nil
+            }
+
+            return FileDiff(parsedHeader: parseHeader(header), hunks: parseHunks(Array(headerAndHunks.dropFirst())))
+        }
+    }
+
+    private func parseHeader(_ header: String) -> FileDiff.ParsedHeader {
+        let lines = header.components(separatedBy: "\n")
+
+        let filePath = lines.first?.split(separator: " ").first(where: { $0.starts(with: "b/") })?.deletingPrefix("b/") ?? ""
```

Comment:
<!--
  0 failure: 
  1 warning:  Line should be 12...
  
  
  DangerID: danger-id-Danger;
  File: Sources/Danger/GitDiff.swift;
  Line: 116;
-->

- :warning: Line should be 120 characters or less: currently 125 characters (`line_length`)